brushfire-core/src/main/scala/com/stripe/brushfire/Injections.scala

```diff
@@ -109,28 +109,35 @@ object JsonInjections {
     }
   }
 
-  implicit def treeJsonInjection[K, V, T](
+  implicit def treeJsonInjection[K, V, T, A](
     implicit kInj: JsonNodeInjection[K],
     pInj: JsonNodeInjection[T],
     vInj: JsonNodeInjection[V],
+    maybeAInj: (Unit =:= A) WithFallback JsonNodeInjection[A],
     mon: Monoid[T],
-    ord: Ordering[V] = null): JsonNodeInjection[Tree[K, V, T]] = {
+    ord: Ordering[V] = null): JsonNodeInjection[AnnotatedTree[K, V, T, A]] = {
 
-    implicit def nodeJsonNodeInjection: JsonNodeInjection[Node[K, V, T, Unit]] =
-      new AbstractJsonNodeInjection[Node[K, V, T, Unit]] {
-        def apply(node: Node[K, V, T, Unit]) = node match {
-          case LeafNode(index, target, _) =>
+    implicit def nodeJsonNodeInjection: JsonNodeInjection[Node[K, V, T, A]] =
+      new AbstractJsonNodeInjection[Node[K, V, T, A]] {
+        def apply(node: Node[K, V, T, A]) = node match {
+          case LeafNode(index, target, annotation) =>
             val obj = JsonNodeFactory.instance.objectNode
             obj.put("leaf", toJsonNode(index))
             obj.put("distribution", toJsonNode(target))
+            maybeAInj.withFallback { aInj =>
```

Comment:
I like the use of `WithFallback` to avoid serializing `Unit` but maybe a comment explaining what is going on (either here or in the method parameters) would be helpful.